### ISSUE TITLE
azurerm_application_insights data resource

### ISSUE THREAD

**Issue by inkel** [terraform-providers/terraform-provider-azurerm#2623]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Hi maintainers, happy new year, and thanks for all your efforts in maintaining this provider.

In the past I've sent a PR (#2463) to add the new `azurerm_virtual_machine` data resource. For the same project I'm working on, I'm kind of in need of having an `azurerm_application_insights` data resource too, are you considering adding it yourselves to the next release or should I send a PR?

As in #2463 the needs arise from wanting to use an Application Insights instrumentation key in another resource (`azurerm_app_service` environment settings). The infra in this project is managed by different teams, and the App Insights bucket I'm trying to read its instrumentation key from it's already created. Now, one possible solution could be to use a `azurerm_application_insights` on my App Service Terraform definition and then `terraform import` the existing resource; sadly this won't scale as the App Insights bucket is being currently managed by hand by another team.

My proposal would be to add a new `azurerm_application_insights` data resource that let me access the instrumentation key by searching for the App Insights bucket name, resource group and location (this last property I'm not sure if it's required).

Thanks.

### New or Affected Resource(s)

* azurerm_application_insights data resource.

### Potential Terraform Configuration

```hcl
// This is the proposed new data resource configuration
data "azurerm_application_insights" "myappinsights" {
    name = "my_app_insights_bucket"
    resource_group_name = "${var.resource_group_name}"
}

// And here we use it to inject the AppInsights' instrumentation key in another resource
resource "azurerm_app_service" "myappsvc" {
    # …all required properties…

    app_settings {
        "APPINSIGHTS_INSTRUMENTATIONKEY" = "${data.azurerm_application_insights.myappinsights.instrumentation_key}"
    }
}
```

### References

* https://www.terraform.io/docs/providers/azurerm/r/application_insights.html

**Comment by tombuildsstuff** [comment]
hey @inkel 

Thanks for opening this issue :)

> In the past I've sent a PR (#2463) to add the new azurerm_virtual_machine data resource. For the same project I'm working on, I'm kind of in need of having an azurerm_application_insights data resource too, are you considering adding it yourselves to the next release or should I send a PR?

We've not got (scheduled) plans to work on this but I'd be 👍 for this if you'd like to send a PR? :)

> My proposal would be to add a new azurerm_application_insights data resource that let me access the instrumentation key by searching for the App Insights bucket name, resource group and location (this last property I'm not sure if it's required).

Based on the API call I believe we should be able to just require the Application Insights name and the Resource Group Name (and not the location) - but that sounds good to me 👍

Thanks!

**Comment by inkel** [comment]
This has been implemented in #2625, so I guess it's safe to close this one.
